core/src/main/java/org/wso2/msf4j/ServiceMethodInfo.java

```diff
@@ -44,6 +47,10 @@ public Method getMethod() {
         return method;
     }
 
+    public Object[] getArgs() {
+        return Arrays.copyOf(args, args.length);
```

Comment:
args[] has a "final" modifier but if we do not use array.copy(), internal objects can be modified by outside.